src/Symfony/Component/Mailer/Transport/Smtp/Stream/SocketStream.php

```diff
@@ -140,6 +140,9 @@ public function initialize(): void
         if ($this->streamContextOptions) {
             $options = array_merge($options, $this->streamContextOptions);
         }
+        if ($this->isTLS()) {
+            $options['ssl']['crypto_method'] = $options['ssl']['crypto_method'] ?? STREAM_CRYPTO_METHOD_TLS_CLIENT;
```

Comment:
this only enables TLS 1.0, which is not good for server compatibility.

Reply:
a better default might be `STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT`  (or maybe without TLS 1.0)

Reply:
see https://www.php.net/manual/fr/function.stream-socket-enable-crypto.php#119122

Reply:
actually, using `STREAM_CRYPTO_METHOD_TLS_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT | STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT` is even better, as `STREAM_CRYPTO_METHOD_TLS_CLIENT` has saner defaults in PHP 7.3+ (https://wiki.php.net/rfc/improved-tls-constants) so this supports both 7.3+ with proper TLS (including TLS 1.3 once supported) but also older versions (by forcing TLS 1.1 and 1.2 support)

Reply:
@stof I totally trust you on that point :simple_smile: